File: java/src/strategy/InsuranceCompany.java
package strategy;

import strategy.evaluators.InsuranceEvaluator;
import strategy.evaluators.InsuranceEvaluatorFactory;
import strategy.models.Vehicle;

public class InsuranceCompany {

    public InsuranceCompany(){}

    public int getQuoteForVehicle(Vehicle vehicle){

        InsuranceEvaluator insuranceEvaluator = InsuranceEvaluatorFactory.createInsuranceEvaluator(vehicle);

        int quote = insuranceEvaluator.getQuoteForVehicle();

        return quote;
    }
}

File: js/src/strategy/models/InsuranceCompany.js
var insuraceEvaluatorFactory = require("../evaluators/InsuranceEvaluatorFactory")();

function InsuraceCompany(){
  return {    
      quoteVehicle: function(vehicle){
           
           // Get Evaluator
           var evaluator = insuraceEvaluatorFactory.createEvaluator(vehicle);
           
           // return quote
           return evaluator.getQuote();
        }
   };
}

module.exports = InsuraceCompany; 
